src/component.js

```diff
@@ -19,27 +19,32 @@ export default {
     };
   },
   created() {
-    this.widgetFactory = this.connector(this.updateData, () => {});
-    this.widget = this.widgetFactory(this.widgetParams);
+    this.factory = this.connector(this.updateState, () => {});
+    this.widget = this.factory(this.widgetParams);
     this.instantSearchInstance.addWidget(this.widget);
   },
   beforeDestroy() {
     this.instantSearchInstance.removeWidget(this.widget);
   },
   watch: {
-    widgetParams(newVal) {
-      const oldWidget = this.widget;
-      this.widget = this.widgetFactory(newVal);
+    widgetParams(nextWidgetParams) {
+      this.state = null;
+      this.instantSearchInstance.removeWidget(this.widget);
```

Comment:
I wonder if this will cause a flash of double applied search parameters?

Reply:
if that was the case, in the previous implementation, there would be a flash of empty parameters

Reply:
The behaviour is the same as before, you can test on a slow network to see what happens.